cluster-autoscaler/core/scale_down.go

```diff
@@ -266,6 +268,11 @@ func (sd *ScaleDown) TryToScaleDown(nodes []*apiv1.Node, pods []*apiv1.Pod, pdbs
 	// to recreate on other nodes.
 	emptyNodes := getEmptyNodes(candidates, pods, sd.context.MaxEmptyBulkDelete, sd.context.CloudProvider)
 	if len(emptyNodes) > 0 {
+		nodeDeletionStart := time.Now()
+		defer func(nodeDeletionStart time.Time, nodeDeletionDuration *time.Duration) {
+			stop := time.Now()
+			*nodeDeletionDuration = stop.Sub(nodeDeletionStart)
+		}(nodeDeletionStart, &nodeDeletionDuration)
```

Comment:
That part feels a bit hacky. There aren't actually any returns from this branch, until the deletion is already done. Why not just update nodeDeletionDuration in line 314 (before if finalError == nil)